functions.php

```diff
@@ -73,65 +114,73 @@ public function register_taxonomies() {
-			'name' => esc_html__( 'Header Title', 'planet4-master-theme' ),
-			'desc' => esc_html__( 'Header title comes here', 'planet4-master-theme' ),
-			'id'   => $prefix . 'title',
-			'type' => 'text_medium',
-		) );
-
-		$p4_header->add_field( array(
-			'name' => esc_html__( 'Header Subtitle', 'planet4-master-theme' ),
-			'desc' => esc_html__( 'Header subtitle comes here', 'planet4-master-theme' ),
-			'id'   => $prefix . 'subtitle',
-			'type' => 'text_medium',
-		) );
-
-		$p4_header->add_field( array(
-			'name'    => esc_html__( 'Header Description', 'planet4-master-theme' ),
-			'desc'    => esc_html__( 'Header description comes here', 'planet4-master-theme' ),
-			'id'      => $prefix . 'description',
-			'type'    => 'wysiwyg',
-			'options' => array(
-				'textarea_rows' => 5,
-			),
+				'id'            => $prefix . 'metabox',
+				'title'         => esc_html__( 'Page Header Fields', 'planet4-master-theme' ),
+				'object_types'  => array( 'page' ), // Post type
 		) );
 
-		$p4_header->add_field( array(
-			'name' => esc_html__( 'Header Button Title', 'planet4-master-theme' ),
-			'desc' => esc_html__( 'Header button title comes here', 'planet4-master-theme' ),
-			'id'   => $prefix . 'button_title',
-			'type' => 'text_medium',
-		) );
-
-		$p4_header->add_field( array(
-			'name' => esc_html__( 'Header Button Link', 'planet4-master-theme' ),
-			'desc' => esc_html__( 'Header button link comes here', 'planet4-master-theme' ),
-			'id'   => $prefix . 'button_link',
-			'type' => 'text_medium',
-		) );
+		$p4_header->add_field(
+			array(
+				'name' => esc_html__( 'Header Title', 'planet4-master-theme' ),
+				'desc' => esc_html__( 'Header title comes here', 'planet4-master-theme' ),
+				'id'   => $prefix . 'title',
+				'type' => 'text_medium',
+			)
+		);
+
+		$p4_header->add_field(
+			array(
+				'name' => esc_html__( 'Header Subtitle', 'planet4-master-theme' ),
+				'desc' => esc_html__( 'Header subtitle comes here', 'planet4-master-theme' ),
+				'id'   => $prefix . 'subtitle',
+				'type' => 'text_medium',
+			)
+		);
+
+		$p4_header->add_field(
+			array(
+				'name'    => esc_html__( 'Header Description', 'planet4-master-theme' ),
+				'desc'    => esc_html__( 'Header description comes here', 'planet4-master-theme' ),
+				'id'      => $prefix . 'description',
+				'type'    => 'wysiwyg',
+				'options' => array(
+					'textarea_rows' => 5,
+				),
+			)
+		);
+
+		$p4_header->add_field(
+			array(
+				'name' => esc_html__( 'Header Button Title', 'planet4-master-theme' ),
+				'desc' => esc_html__( 'Header button title comes here', 'planet4-master-theme' ),
+				'id'   => $prefix . 'button_title',
+				'type' => 'text_medium',
+			)
+		);
+
+		$p4_header->add_field(
+			array(
+				'name' => esc_html__( 'Header Button Link', 'planet4-master-theme' ),
+				'desc' => esc_html__( 'Header button link comes here', 'planet4-master-theme' ),
+				'id'   => $prefix . 'button_link',
+				'type' => 'text_medium',
+			)
+		);
```

Comment:
All registered fields in this file do not need escaping, only translation. We escape on output, in this case we're relying on shortcode to handle escaping for us.

Reply:
Can you please explain better ? What do you suggest me to do here?

Reply:
Sure, it's similar to https://github.com/greenpeace/planet4-plugin-blocks/pull/1#discussion_r141006483

We're escaping late, i.e. on direct output of strings, usually where there are some printing functions like `echo` or `printf()`. If we register a translation string that is then processed by other function (via arguments or any other way) we trust that function will do the correct escaping output.

So in here, you're registering strings for translation, but you're not printing them here directly, but passing on to another function. So you only need to register translation here, without escaping:
```
$p4_header->add_field(
		array(
				'name' => __( 'Header Button Link', 'planet4-master-theme' ),
				'desc' => __( 'Header button link comes here', 'planet4-master-theme' ), 
				'id'   => $prefix . 'button_link',
				'type' => 'text_medium',
		)
);
```

P.S. I'd suggest these articles to read up on translation and escaping:
https://codex.wordpress.org/I18n_for_WordPress_Developers
http://ottopress.com/2012/internationalization-youre-probably-doing-it-wrong/